### PR TITLE
General Updates 11

### LINKED ISSUE #8: Add multi-account support
### Before you start...

- [X] Have you updated the FluentHub? You might be using an old version.
- [X] Have you checked if similar issue have already been reported?

### What windows are you seeing the problem on?

Windows 11

### What FluentHub version/architecture are you seeing the problem on?

_None_

### Description

As the title

### Steps To Reproduce

_No response_

### Expected behavior

_No response_

### Relevant Assets

_No response_

### PR DESCRIPTION
**Pull request by BobbyESP**
### Before you start...

- [x] Have you updated the FluentHub? You might be using an old version.
- [x] Have you checked if similar issue have already been reported?

### Resolved / Related Issues
- Closes #8 (Wow, the 8th issue haha)

### Details of Changes

- Added the ability to switch between different GitHub accounts without singing out.

### Validation
In the appbar click on the 3 dots icon and in that flyout click on "Switch Account"

- [x] Built and ran the app
- [x] Tested the changes for accessibility

### Relevant Assets
image

### CODE REVIEW

**Review by onein528 — DISMISSED** [review]
Incredible idea! LGTM.

**Comment by onein528** [comment]
it has difficult. we can store several account info, but browser has cookie or browser data. so once user authorize app with one account, user is hard to authorize again with other account.

**Comment by BobbyESP** [comment]
True. Maybe, the best way is trying to find a way to open a private tab so they canclogin with the account they want

**Comment by onein528** [comment]
yep. i think so

**Comment by onein528** [comment]
so app would display auth link and indicate user to open private tab. UWP app doesn't have capability to open specified browser.

**Comment by BobbyESP** [comment]
Yes, that's one idea, but maybe it's better to use WebView window/view for login?

**Comment by onein528** [comment]
better to use webview2
but it seems to remain browser data

**Comment by BobbyESP** [comment]
Yeah, we had to implement a clean data programatically

**Comment by BobbyESP** [comment]
And i need to take a look to that because WebView2 was in Beta version and apps with it can't be uploaded to the store

**Comment by onein528** [comment]
WebView2 is very nice idea!(not needed to consider about user's cookies, caches, and more)
We can update to pre-release version of muxc.

> apps with it can't be uploaded to the store
Yes. This is the most important point on this issue. maybe we have to wait until publish stable version of WebView2, means WinUI 2.8

**Comment by BobbyESP** [comment]
Ok, nice!

**Comment by onein528** [comment]
However we can still have WebView as another option. I need to take a look.

**Comment by BobbyESP** [comment]
We should take a look to this docs. Maybe now we can be able to upload to the store without problems. I don't really know, the thing is to test it, but now i'm at school.

https://docs.microsoft.com/en-us/microsoft-edge/webview2/how-to/publish-uwp-app-store

**Comment by onein528** [comment]
Previously, we had this problem. I dont know why, despite I had latest SDK.

> Supported API
In this test, WACK checks the Windows APIs that your host app is calling, to make sure they're supported.</br>If your app failed the Supported API test, update the Windows SDK to version 22000 or higher, to get the latest version of the allowed-APIs list. There should be no blocking APIs for WebView2 in the current Windows SDK.

**Comment by onein528** [comment]
But I can try again.

**Comment by BobbyESP** [comment]
> But I can try again.
> 
> I have many exams now, so I will do after another 2 weeks.

Same...

**Comment by onein528** [comment]
back off. @stale

**Comment by onein528** [comment]
I can work on this!

**Comment by BobbyESP** [comment]
Ok nice! Go with it!

**Comment by onein528** [comment]
image

Nice

**Comment by onein528** [comment]
Like? @BobbyESP 

image

**Comment by BobbyESP** [comment]
Ohh yes! Cool men!

**Comment by onein528** [comment]
Sign in with WebView2 window

image

**Comment by onein528** [comment]
But could not redirect. so we need to instruct users
1. Show url in the textbox
2. Paste In-Private url bar
3. Go there
4. And get back

**Comment by onein528** [comment]
I dont know if it is better

image

**Comment by BobbyESP** [comment]
Wooow it's nice meen!!

**Comment by onein528** [comment]
Do not re-authorize or switch user. latest commit makes the app go crazy.

**Comment by onein528** [comment]
Niceee!!!

**Comment by onein528** [comment]
How about? @BobbyESP @DeveloperWOW64 

image

**Comment by BobbyESP** [comment]
Woow, that's nice! I like it haha

**Comment by onein528** [comment]
I will definitely make this great, so look forward to it!

**Comment by BobbyESP** [comment]
Okeyy, nice men! Tomorrow i finish my exams (well, i finish secondary school in Spain(12-16 years))

**Comment by DeveloperWOW64** [comment]
> Okeyy, nice men! Tomorrow i finish my exams (well, i finish secondary school in Spain(12-16 years))

😀😀😀 That's amazing!!
 Make sure to have lots of sleep - as one of my teachers Mrs Legg once said to my class "I'd rather you spent your time sleeping than staying up studying". 
I'm not done yet with school - though I did recently finish some large exams. Do you have A-levels in Spain?

**Comment by BobbyESP** [comment]
Thanks men! And  yes, maybe i should sleep more haha... These days are killing me, sleeping 6h/day and the rest of the day, studying. 

And no, we don't have A-Levels

**Comment by onein528** [comment]
A-Level is only applicable for admission to universities in the UK and its nations, maybe. @DeveloperWOW64

**Comment by onein528** [comment]
can I re-open this as General Updates 10? you'll be a co-author, instead.

**Comment by DeveloperWOW64** [comment]
> can I re-open this as General Updates 10? you'll be a co-author, instead.

WDYM? or are you talking to BobbyESP?

**Comment by onein528** [comment]
Sorry yes. @BobbyESP

**Comment by BobbyESP** [comment]
> can I re-open this as General Updates 10? you'll be a co-author, instead.

Yeah, no problem

**Comment by onein528** [comment]
DONE. All things should work.

**Comment by onein528** [comment]
but account switching still is not done.

**Comment by BobbyESP** [comment]
No problem!

**Comment by DeveloperWOW64** [comment]
> but account switching still is not done.

If account switching isn't done, what have else have you finished?

**Comment by onein528** [comment]
take a loo at latest commit. I completely changed actions when navigated to the page. previously, at that time, app passed RepoContextViewModel or sth. but in considering url navigation, i changed to pass the page's url.

**Comment by onein528** [comment]
And details layout page's code behind had very ugly code and so improved easier to see.

**Comment by DeveloperWOW64** [comment]
OH WOW! It looks amazing!

**Comment by onein528** [comment]
## What's new
c43ff60 Now, you can click issue or pull button and go to the page from like this page:

image

**Comment by onein528** [comment]
I am planning to implement the feature of pagenation using Incremental Loading on GU 12 or later.

**Comment by DeveloperWOW64** [comment]
> I am planning to implement the feature of pagenation using Incremental Loading on GU 12 or later.

WDYM?

**Comment by onein528** [comment]
Cirrently, the app shows only 20 items and cannot load more items. so I want to implement loading more items using Incremental Loading which is one of the ms community toolkit.

**Comment by DeveloperWOW64** [comment]
Right. This'll take some work:
image

**Comment by onein528** [comment]
oh my... after commited uncommited stuff, i will take a look....

**Comment by DeveloperWOW64** [comment]
@onein528 Just finished resolving the conflicts 🎉🎉

**Comment by DeveloperWOW64** [comment]
@onein528 I'm still used to the old param method:
image
how do the viewmodels work?

**Comment by onein528** [comment]
> @onein528 I'm still used to the old param method: image how do the viewmodels work?

I am working on that. and likely to fix.

**Comment by onein528** [comment]
@DeveloperWOW64 Fixed all.

File: src/FluentHub/Views/AppSettings/MainSettingsPage.xaml [hunk 1]
```diff
@@ -105,19 +105,19 @@
...
-                        <muxc:ImageIcon Source="/Assets/EmojiImages/repository.png" />
+                        <muxc:ImageIcon Source="/Assets/Icons/Repositories.png" />
                     </muxc:NavigationViewItem.Icon>
                 </muxc:NavigationViewItem>
 
                 <muxc:NavigationViewItem Content="Notifications" Tag="notifications">
                     <muxc:NavigationViewItem.Icon>
-                        <muxc:ImageIcon Source="/Assets/EmojiImages/appointment-reminders.png" />
+                        <muxc:ImageIcon Source="/Assets/Notifications.png" />
                     </muxc:NavigationViewItem.Icon>
```
Comment (onein528) [inline src/FluentHub/Views/AppSettings/MainSettingsPage.xaml:121]: mistake?

Comment (onein528) [reply src/FluentHub/Views/AppSettings/MainSettingsPage.xaml:121]: should

/Assets/Icons/Notifications.png

Comment (DeveloperWOW64) [reply src/FluentHub/Views/AppSettings/MainSettingsPage.xaml:121]: Not mistake - I'm unifying the icons to only have one size and in the same folder

File: src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs [hunk 2]
```diff
@@ -29,10 +29,6 @@ protected override void OnNavigatedTo(NavigationEventArgs e)
             var currentItem = navigationService.TabView.SelectedItem.NavigationHistory.CurrentItem;
             currentItem.Header = "Insights";
             currentItem.Description = "Insights";
-
-            currentItem.Url = url;
-            currentItem.DisplayUrl = $"{pathSegments[0]} / {pathSegments[1]} / Insights";
-
             currentItem.Url = $"{url}";
             currentItem.DisplayUrl = $"{pathSegments[3]} / {pathSegments[4]} / Insights";
```
Comment (onein528) [inline src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs:34]: Dont change.....

Comment (DeveloperWOW64) [reply src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs:34]: Okay - wait you mean turn it back to [0] [1]?

File: src/FluentHub/Views/AppSettings/MainSettingsPage.xaml (hunk 1 above)
Comment (onein528) [reply src/FluentHub/Views/AppSettings/MainSettingsPage.xaml:121]: but notifications.png in `/Assets/Icons/Notifications.png` not in `/Assets/Notifications.png`

Comment (DeveloperWOW64) [reply src/FluentHub/Views/AppSettings/MainSettingsPage.xaml:121]: Fixed on my machine - let me push

Comment (DeveloperWOW64) [reply src/FluentHub/Views/AppSettings/MainSettingsPage.xaml:121]: done

File: src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs (hunk 2 above)
Comment (onein528) [reply src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs:34]: yep.

Comment (onein528) [reply src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs:34]: pathSegments starts with like FluentHub/FluentHub/pulse

**Comment by onein528** [comment]
> Okay this icon thing is kinda getting ridiculous..

WDYM?

File: src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs (hunk 2 above)
Comment (DeveloperWOW64) [reply src/FluentHub/Views/Repositories/Insights/InsightsPage.xaml.cs:34]: You want me to change it from / Insights to / Pulse?
Personally Insights is easier to understand than Pulse

**Comment by DeveloperWOW64** [comment]
> > Okay this icon thing is kinda getting ridiculous..
> 
> WDYM?

image

**Comment by onein528** [comment]
hahaha right

**Comment by DeveloperWOW64** [comment]
@onein528 Maybe this is debatable but can I rename the Accounts settings page to Account? Multi-account support is only a sub-feature of the accounts page and isn't entirely the account page's purpose.

**Comment by onein528** [comment]
> I rename the Accounts settings page to Account

Where is it?

**Comment by DeveloperWOW64** [comment]
> > I rename the Accounts settings page to Account
> 
> Where is it?

Nowhere. I'm asking if it's okay or you want to keep it as Accounts

**Comment by onein528** [comment]
ahh, yes! can change it.

**Comment by DeveloperWOW64** [comment]
Amazing!

**Comment by onein528** [comment]
i will sleep bye.

**Comment by DeveloperWOW64** [comment]
> i will sleep bye.

👋 good night

**Comment by onein528** [comment]
Nice work!!

**Comment by onein528** [comment]
@DeveloperWOW64 Committed.

**Comment by DeveloperWOW64** [comment]
> @DeveloperWOW64 Committed.

thanks!

**Comment by onein528** [comment]
i got something bad. i will fix and commit again. @DeveloperWOW64

**Comment by DeveloperWOW64** [comment]
> i got something bad. i will fix and commit again. @DeveloperWOW64

`<DataTemplate  x:DataType="x:String" />`
This?

**Comment by onein528** [comment]
yes. committed again.

**Comment by BobbyESP** [comment]
Guys, i think i can came back to work!

**Comment by onein528** [comment]
woow. we are team, right? very exited to work together.

**Comment by DeveloperWOW64** [comment]
> woow. we are team, right? very exited to work together.

yes 😀

**Pull request closed without merging**